main/realmain.cc

```diff
@@ -483,6 +483,9 @@ int realmain(int argc, char *argv[]) {
     if (opts.noErrorSections) {
         gs->includeErrorSections = false;
     }
+    if (opts.disableRuby3Numparams) {
+        gs->checkReservedNumberedParameters = false;
+    }
```

Comment:
@froydnj brought up a good point today, which is that we could use the existing `--suppress-error-code` option to disable this error if it had a unique identifier. What do you think about adding a custom error to `core/errors/parser.h` for the numparams error, and then recommending using `--suppress-error-code` with that number in the error message instead?

Reply:
I like that idea better indeed. Changed the code to use a new error class in the latest commit.